mart/attack/perturber/perturber.py

```diff
@@ -46,7 +46,9 @@ def __init__(
         self.projector = projector
         self.optim_params = optim_params
 
-        self.perturbation = torch.nn.UninitializedParameter()
+        # Register perturbation as a non-persistent buffer even though we will optimize it. This is because it is not
+        # a parameter of the underlying model but a parameter of the adversary.
+        self.register_buffer("perturbation", torch.nn.UninitializedParameter(), persistent=False)
```

Comment:
Training on GPU would fail: `python -m mart experiment=CIFAR10_CNN_Adv trainer=gpu`

I guess `model.cuda()` materializes the uninitialized buffer?